plugins/cameraserver/src/main/java/edu/wpi/first/shuffleboard/plugin/cameraserver/source/CameraUrlGenerator.java

```diff
@@ -45,31 +51,95 @@ public CameraUrlGenerator(CameraServerSource source) {
     if (frameRate > 0) {
       commands.put("fps", Integer.toString(frameRate));
     }
-    return generateUrls(commands, baseUrls);
+    // Add a special case command for NI Cameras
+    if (source.getName().contains("IMAQdx")) {
+      commands.put("name", source.getName());
+    }
+    return generateUrls(commands, baseUrls, source.getName());
   }
 
   @VisibleForTesting
-  static String[] generateUrls(Map<String, String> commands, String[] baseUrls) { // NOPMD varargs instead of array
+  static String[] generateUrls(Map<String, String> commands, String[] baseUrls,
+                               String cameraName) {
     if (baseUrls == null || baseUrls.length == 0) {
       return new String[0];
     }
     if (commands.isEmpty()) {
       return baseUrls;
     } else {
-      return Arrays.stream(baseUrls)
-          .map(url -> url + toHttpParams(commands))
+      var urls =  Arrays.stream(baseUrls)
+          .map(url -> addHttpParams(url, commands))
+          .filter(url -> url != null)
```

Comment:
Use the `Objects::nonNull` method reference

Reply:
Fixed